### PR TITLE
Add documentation.

### PR DESCRIPTION
**Pull request by jesnie**
Fixes: #12

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = 'alabaster'
+
+# Add any paths that contain custom static files (such as style sheets) here,
+# relative to this directory. They are copied after the builtin static files,
+# so a file named "default.css" will overwrite the builtin "default.css".
+html_static_path = ['_static']
```
Comment (uri-granta) [inline docs/conf.py:70]: FYI I managed to get a version navbar for [trieste](https://secondmind-labs.github.io/trieste/0.12.0/index.html) without _too_ much pain. Also, a github link.

File: .github/workflows/build_test_deploy.yml [hunk 2]
```diff
@@ -21,30 +23,58 @@ jobs:
...
+        run: |
+          branch=$(git symbolic-ref --short HEAD)
+          rm -rf gh-pages/docs/${branch}
+          mv docs_build gh-pages/docs/${branch}
+          cd gh-pages
+          git add docs
+          git config --global user.email "none"
+          git config --global user.name "github-actions-bot"
+          git commit -m "Build documentation."
+          git push
```
[the remark this answers is not in the document]
Comment (jesnie) [reply .github/workflows/build_test_deploy.yml:76]: > Does the documentation include generated images? If so, might it be better to force do a `git commit --amend --no-edit` when deploying new `develop` docs, to overwrite the previous commit and avoid making the repository grow continuously.

Right now the documentation doesn't include images at all. :shrug:

File: docs/conf.py (hunk 1 above)
Comment (jesnie) [reply docs/conf.py:70]: > FYI I managed to get a version navbar for [trieste](https://secondmind-labs.github.io/trieste/0.12.0/index.html) without _too_ much pain. Also, a github link.

Yeah, GPflow has the same navbar version switcher. However, it is currently broken. Apparently the PyData people aren't very good at backwards compatibility and right now I don't have the energy to deal with that. Maybe another day.
https://github.com/pydata/pydata-sphinx-theme/issues/932

File: docs/index.rst [hunk 3]
```diff
@@ -0,0 +1,471 @@
+check_shapes User Guide
+=======================
+
+A library for annotating and checking the shapes of tensors.
```
[the remark this answers is not in the document]
Comment (jesnie) [reply docs/index.rst:4]: > Maybe mention some of the features here? E.g. that it rewrites docstrings, supports named variable specifications, etc?

Done. I've also added an "installation" section.

**Pull request merged**